__tests__/app.test.js

```diff
@@ -81,6 +82,58 @@ describe('/api/articles/:article_id', () => {
                 expect(response.body.msg).toBe('bad request');
             });
     });
+    test('PATCH:200 updates article (increments votes) in db and sends updated article back to client', () => {
+        const updatedArticle = { inc_votes: 2 };
+        return request(app)
+            .patch('/api/articles/3')
+            .send(updatedArticle)
+            .expect(200)
+            .then((response) => {
+                const { article } = response.body;
+                expect(article.votes).toBe(2);
+            });
+    });
+    test('PATCH:200 does not decrement votes to value below 0', () => {
+        const updatedArticle = { inc_votes: -20 };
+        return request(app)
+            .patch('/api/articles/3')
+            .send(updatedArticle)
+            .expect(200)
+            .then((response) => {
+                const { article } = response.body;
+                expect(article.votes).toBe(0);
+            });
+    });
+    test('PATCH:400 sends error message when given a valid but non-existent article id', () => {
+        const updatedArticle = { inc_votes: 2 };
+        return request(app)
+            .patch('/api/articles/99')
+            .send(updatedArticle)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('bad request');
+            });
+    });
+    test('PATCH:400 sends error message when given an invalid article id', () => {
+        const updatedArticle = { inc_votes: 2 };
+        return request(app)
+            .patch('/api/articles/not-a-number')
+            .send(updatedArticle)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('bad request');
+            });
+    });
+    test('PATCH:400 sends error message when request body contains value of incorrect type', () => {
+        const updatedArticle = { inc_votes: 'not-a-number' };
+        return request(app)
+            .patch('/api/articles/3')
+            .send(updatedArticle)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('bad request');
+            });
+    });
```

Comment:
You could also test for missing properties which would be a 400 also, however this can be added later and should be considered an extra.